src/Rule/LatteTemplatesRule.php

```diff
@@ -71,28 +81,49 @@ public function processNode(Node $collectedDataNode, Scope $scope): array
         foreach ($this->latteTemplateResolvers as $latteTemplateResolver) {
             foreach ($resolvedNodeFinder->find(get_class($latteTemplateResolver)) as $collectedResolvedNode) {
                 $templates = $latteTemplateResolver->findTemplates($collectedResolvedNode, $collectedDataNode);
-                foreach ($templates as $template) {
-                    $templatePath = $template->getPath();
-
-                    try {
-                        $compileFilePath = $this->latteToPhpCompiler->compileFile($template->getActualClass(), $templatePath, $template->getVariables(), $template->getComponents());
-                    } catch (Throwable $e) {
-                        $errors = array_merge($errors, $this->errorBuilder->buildErrors([new Error($e->getMessage(), $scope->getFile())], $templatePath, $scope));
-                        continue;
-                    }
-
-                    $fileAnalyserResult = $this->fileAnalyserFactory->create()->analyseFile(
-                        $compileFilePath,
-                        [],
-                        $this->rulesRegistry,
-                        $this->collectorsRegistry,
-                        null
-                    );
-
-                    $errors = array_merge($errors, $this->errorBuilder->buildErrors($fileAnalyserResult->getErrors(), $templatePath, $scope));
-                }
+                $this->analyseTemplates($templates, $scope, $errors);
             }
         }
         return $errors;
     }
+
+    /**
+     * @param Template[] $templates
+     * @param RuleError[] $errors
+     */
+    private function analyseTemplates(array $templates, Scope $scope, array &$errors): void
+    {
+        foreach ($templates as $template) {
+            $templatePath = $template->getPath();
+
+            try {
+                $compileFilePath = $this->latteToPhpCompiler->compileFile($template->getActualClass(), $templatePath, $template->getVariables(), $template->getComponents());
+            } catch (Throwable $e) {
+                $errors = array_merge($errors, $this->errorBuilder->buildErrors([new Error($e->getMessage(), $scope->getFile())], $templatePath, $scope));
+                continue;
+            }
+
+            $fileAnalyserResult = $this->fileAnalyserFactory->create()->analyseFile(
+                $compileFilePath,
+                [],
+                $this->rulesRegistry,
+                new CollectorsRegistry([$this->includePathCollector]),
```

Comment:
because we don't need all of them, just include path collector here. It is some micro optimization - there are no method calls, variables, etc analysed on compiled template